rust/tools/authorization-logic/src/parsing/AuthLogic.g4

```diff
@@ -139,7 +145,9 @@ ATTRIBUTE: 'attribute';
 
 // Identifiers wrapped in quotes are constants whereas
 // identifiers without quotes are variables.
-ID : ('"')? [_a-zA-Z][_a-zA-Z0-9/.#:]* ('"')?;
+VARIABLE : [_a-zA-Z][_a-zA-Z0-9/.#:]*; 
+CONSTANT : '"' [_a-zA-Z+-][@_a-zA-Z0-9/.#:+=-]* '"';
```

Comment:
I just noticed this - you're adding `+` and `-` as potential first characters in the pattern for `CONSTANT`. I think what you're trying to do is to say that a constant can start with a `+` or a `-`, indicating a unary plus or minus. But you're entangling that with the concern of "allowing a smaller set of characters to be the first character of an identifier" - which means that while `"0"` is not a legal constant, `"-0"` is. Of course, I'm not entirely clear, mentally, what a constant is in this context, so maybe this is ok.

I do wonder about that first-character restriction for constants at all though. I get it for identifiers, it's a common way to keep from confusing humans reading code, but I'm not sure what function it serves for constants which will be wrapped in `"` or `"""` anyway.

Reply:
>  I think what you're trying to do is to say that a constant can start with a + or a -, indicating a unary plus or minus.

Good guess, but this was actually not meant to be a proxy for unary operators. The only unary operator in this language is negation of predicates and that is handled "the right way" by making this an optional char in the parse rule for predicate and storing negation as part of the AST node for predicates. The actual motivation for including `+-=` in CONSTANT was to support principal names that are public keys like [this one](https://github.com/project-oak/transparent-release/blob/2f25e7a46ea571f7ff33d1e0c35dc279247daadd/experimental/auth-logic/endorsement-release/input_policy_examples/oak_endorsement_policy.auth_logic#L27). But actually, cryptographic keys are going to be MULTILINE_CONSTANTs in all cases, so these characters don't need to be in the rule for CONSTANTs which cannot take multiple lines. I left `-` in the middle of CONSTANT because hyphens do show up in desired principal names for transparent release and this seems OK to support in general IMO.

> I do wonder about that first-character restriction for constants at all though.

The difference in first characters vs other characters in these lex rules exists because I started these from standard lex rules for identifiers and then expanded them as needed from the use-cases I was looking at. I think it's fine to re-evaluate if this is how it should work for our policy language going forward. Since this kind of change is really "shallow" to make (it only requires changing the grammar rules and will have no affect on the data structures implied by the grammar so it will not require any changes to the AST nodes or compiler passes internally). I think it may be best to do this kind of language re-evaluation post migration and as a part of our re-evaulation of our policy language design.